src/server/game/Spells/Spell.cpp

```diff
@@ -5734,15 +5734,8 @@ SpellCastResult Spell::CheckCast(bool strict)
                         if (!target)
                             return SPELL_FAILED_BAD_TARGETS;
 
-                        Position pos;
-                        target->GetChargeContactPoint(m_caster, pos.m_positionX, pos.m_positionY, pos.m_positionZ);
-
                         if (m_caster->GetMapId() == 618) // pussywizard: 618 Ring of Valor
-                            pos.m_positionZ = std::max(pos.m_positionZ, 28.28f);
-
-                        float maxdist = m_caster->GetMeleeRange(target);
-                        if (!target->IsInDist(&pos, maxdist))
-                            return SPELL_FAILED_NOPATH;
+                            target->m_positionZ = std::max(target->m_positionZ, 28.28f);
 
                         if (m_caster->GetMapId() == 618) // pussywizard: 618 Ring of Valor
```

Comment:
Move it under this check then, no need to have double `m_caster->GetMapId() == 618` checks

Reply:
That was something I was looking at, but I would rather just remove these if possible. Looking at Trinity core, they don't have these checks.

Reply:
If those hackfixes are no longer needed then I'd say it is fine imo, but you would have to test it out. (Before and after)